src/main/kotlin/Github.kt

```diff
@@ -191,15 +192,42 @@ class GithubService(config: GithubConfig) {
      * @param pull the pull request for which the statuses are being determined
      */
     fun assessStatusAndChecks(pull: Pull) {
+        val required = getRequiredStatusAndChecks(pull)
+
         val statusCheck = getStatusOrChecks<Check>(pull, SummaryType.CHECK_RUNS) ?: return
         val status = getStatusOrChecks<Status>(pull, SummaryType.STATUS) ?: return
 
-        if (statusCheck.checkRuns.any { checkFailure(it.conclusion) }) {
-            removeLabels(pull, LabelRemovalReason.STATUS_CHECKS)
-        } else if (status.state == "failure" || status.state == "error") {
-            removeLabels(pull, LabelRemovalReason.STATUS_CHECKS)
-        } else if (checksCompleted(statusCheck) && statusesCompleted(status)) {
+        val checks = statusCheck.checkRuns.filter { required.contains(it.name) }.map { it.name to checkState(it) }
+        val statuses = status.statuses.filter { required.contains(it.context) }.map { it.context!! to statusState(it) }
+        val knownStatuses = checks.union(statuses).toMap()
+        val unknownStatuses = required.filter { !knownStatuses.keys.contains(it) }.map { it to StatusState.PENDING }
+        val statusMap: Map<String, StatusState> = checks.union(statuses).union(unknownStatuses).toMap()
```

Comment:
This chunk of code is kind of confusing I think. I would like a second set of eyes on it. Essentially what I'm trying to do is:

I have a list of required checks (Strings with the names of the checks)
`[ "Check1", "Check2" ]`

I have a list of checks and statuses from github (two different object structures)
```
{
  checkRuns: [ { name: "Check1" }, { name: "Foo" } ]
}
```
```
{
  statuses: [ { context: "Check2" }, { context: "Bar" }]
}
```
I want to go through all of the required items, find them in the appropriate checks/statuses object, and create a mapping of `Name` to `StatusState`. Any items that don't exist in the checks/status objects will have the state set to `PENDING` by default. 

What I have hear _works_ but it looks a little hard to follow. @tednology any thoughts?